packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.types.ts

```diff
@@ -313,10 +314,36 @@ export interface UsePickerValueLayoutResponse<TValue extends PickerValidValue>
   isValid: (value: TValue) => boolean;
 }
 
+/**
+ * Params passed to `usePickerProvider`.
+ */
+export interface UsePickerValueProviderParams<TValue extends PickerValidValue> {
+  value: TValue;
+  contextValue: UsePickerValueContextValue;
+}
+
 export interface UsePickerValueResponse<TValue extends PickerValidValue, TError> {
   open: boolean;
   actions: UsePickerValueActions;
   viewProps: UsePickerValueViewsResponse<TValue>;
   fieldProps: UsePickerValueFieldResponse<TValue, TError>;
   layoutProps: UsePickerValueLayoutResponse<TValue>;
+  provider: UsePickerValueProviderParams<TValue>;
+}
+
+export interface UsePickerValueContextValue {
+  /**
+   * Sets the current opening status of the picker.
```

Comment:
Nitpick - feel free to ignore if you feel like `opening status` has a better "ring". 🙈 
```suggestion
   * Sets the current open state of the Picker.
```